### PR TITLE
Feature/calendar

### PR DESCRIPTION
**Pull request by capetillo**
## FEATURE/FIX: Using same calendar when booking RTI and requesting observations

**Background:**
More than one style of calendar was being used and that was because I couldn't find a date picker and a date range picker but now I've found an older version of a package and it works well. 

**Implementation:**
- Initialized the component on `main.js`
- Changed the component in `TimePicker` (that one was pretty simple, not much changed)
- Tweaked a little bit of logic in `BeginnerScheduling` and `AdvancedScheduling` because the component `VDatePicker` contains the object keys of `start` and `end` from the package.
- Added `calendarUtils` but have to discuss with Edward how these should be used. Basically, this gets the start and end dates of the current semester and its purpose is meant to set limits
- And added and updated tests

### VISUALS
**Calendar when booking for RTI**
Screenshot 2024-11-12 at 1 40 22 PM

**Calendar when requesting a scheduled observation (end of current semester is Jan 31)**

https://github.com/user-attachments/assets/35808fa6-d1a6-4ad9-ac6b-ff38e9079c92

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -28,7 +29,7 @@
     "pinia-plugin-persistedstate": "^3.2.1",
     "sass": "^1.75.0",
     "sass-loader": "^14.2.1",
-    "v-calendar": "^3.1.2",
+    "v-calendar": "^3.0.0",
```
Comment (jnation3406) [inline package.json:32]: Did they remove the date range functionality in newer versions or something?

File: src/components/RealTimeInterface/TimePicker.vue [hunk 2]
```diff
@@ -196,15 +199,6 @@ async function getAvailableTimes () {
   })
 }
 
-// Used to block out dates that are not in the availableTimes object from the date picker
-const isDateAllowed = (date) => {
```
Comment (jnation3406) [inline src/components/RealTimeInterface/TimePicker.vue:200]: Is this logic of blocking out dates reimplemented somewhere else now?

**Review by jnation3406 — APPROVED** [review]
Looks good being the same calendar. Just a little confused on the downgrading library version part...

File: package.json (hunk 1 above)
Comment (capetillo) [reply package.json:32]: I'm looking through my history to try to find the github issue but basically the bug didn't take `max-date` into consideration

**Review by mgdaily — APPROVED** [review]

**Pull request merged**